core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3227,6 +3240,28 @@ public void validateIntervalQualifier(SqlIntervalQualifier qualifier) {
     }
   }
 
+  @Override public void validateCreateFunctionSqlForm(
+      SqlCreateFunctionSqlForm createFunction) {
+    Preconditions.checkArgument(createFunction != null);
+    nodeToTypeMap.put(createFunction, unknownType);
```

Comment:
I think the usual behavior for validation is to call setValidatedNodeType rather than writing to the nodeToTypeMap directly.

Reply:
I don't think that's possible for this case since setValidatedNodeType doesn't allow the setting of unknownType

Reply:
Hmm ok, why is unknownType being inserted instead of returnType?

Reply:
I guess it's a question of semantics, I feel like a "CREATE" statement doesn't really have an inherent type. I don't mind changing it to returnType though.

Reply:
Oh, that's fair, CREATE doesn't necessarily have the return type of the function it is defining